### ISSUE TITLE
Pressing a button while pointing both lasers at it spams the button

### ISSUE THREAD

**Issue by ClimberJ** [Yellow-Dog-Man/Resonite-Issues#24]
### Describe the bug?

If you point both of your lasers at a button and hold the trigger one one of your controllers, the button input will be spammed. This seems to happen with all UI buttons. (Tested with keyboard and Flux impulse call)

### To Reproduce

Point both of your lasers at a button and hold the trigger one one of your controllers, the button will be repeatedly triggered.

### Expected behavior

Buttons should behave the same way, independetly of wether or not both lasers are pointing at it.

### Resonite Version Number

2023.10.13.743

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Valve Index

### Reporters

 Climber

**Comment by ExoTheWicker** [comment]
This was also a bug in the old platform as well. I just wanted to put this here incase people are wondering if this is a new problem or if it has existed for a while.

**Comment by Psychpsyo** [comment]
This happens for all UIX canvases.
(~~and I love it, please keep it~~)

**Comment by AmasterAmaster** [comment]
I may wish for it to be removed (at least for some buttons and UI) because it is way too easy for someone to aim at the invite button and spam invite a user. This has happened way more than I like to see as people like to abuse that. Also probably have it removed for keyboard inputs as typing is a bit of a pain when you want to rest your lasers on individual keys and sometimes this bug triggers it.

**Comment by epicEaston197** [comment]
This is really annoying for desktop users as well if two desktop users are staring at a video player or a desktop user in combination with a vr user it results in the video player playing then pausing because of the second user

**Comment by 3x1t-5tyl3** [comment]
> This is really annoying for desktop users as well if two desktop users are staring at a video player or a desktop user in combination with a vr user it results in the video player playing then pausing because of the second user

Just thought I'd mention it here quickly if you wanted a fix in the meantime: make the "hover" bool local. That should fix that issue.

That said I'd love to keep the ability to spam UI in a fun way; I'd love for this as a bugfix but also as an option to "allowSpamming" as by now it's a bug that basically also has become a feature..

**Comment by Redd56** [comment]
is this concidered a feature now or is uix spamming still concidered a bug. cause like i consider it a hidden feature caused by a bug like minecrafts quaziconnectivity

**Comment by Dusty-Sprinkles** [comment]
I agree that this is more annoying than anything, and if it isn't complicated to fix I'd rather have this fixed without needing to implement some sort of replacement and wait for that first.

**Comment by JackTheFoxOtter** [comment]
UIX spamming is absolutely a bug. And a very annoying one at that. It makes creating "multi-touch" interactions (where you can use both lasers on the same button) pretty annoying to implement.

It's also way too jank currently to be considered a feature. I know some applications have buttons that when held for long enough will go into "rapidfire" mode, but if that's something you want, you can already easily implement this in ProtoFlux (we have the "button held" impulse you can use to implement this). You could make this a feature request to be an option on buttons directly, but that's separate from this report.
